### ISSUE TITLE
23.4.0 Free gets "ORA-07427: spstp: cannot change directory to dbs." error

### ISSUE THREAD

**Issue by yahonda** [oracle/docker-images#2801]
23.4.0 Free gets "ORA-07427: spstp: cannot change directory to dbs." error

### Steps to reproduce
```
git clone https://github.com/oracle/docker-images
cd docker-images/OracleDatabase/SingleInstance/dockerfiles
./buildContainerImage.sh -v 23.4.0 -f
sudo docker run -d --name oracle-container  oracle/database:23.4.0-free
```

### Expected behavior
It should start up 23.4.0 Free Docker process.

### Actual behavior
It gets "DATABASE SETUP WAS NOT SUCCESSFUL!".

```
$ docker logs -f 9e2ad8cdd05889c8b7fa4c2b17a377a6ff7ec252625d7777b84a0b1f8e765053
Specify a password to be used for database accounts. Oracle recommends that the password entered should be at least 8 characters in length, contain at least 1 uppercase character, 1 lower case character and 1 digit [0-9]. Note that the same password will be used for SYS, SYSTEM and PDBADMIN accounts:
Confirm the password:
Configuring Oracle Listener.
Listener configuration succeeded.
Configuring Oracle Database FREE.
Enter SYS user password:
*****************
Enter SYSTEM user password:
****************
Enter PDBADMIN User Password:
*****************
Prepare for db operation
7% complete
Copying database files
29% complete
Creating and starting Oracle instance
30% complete
33% complete
36% complete
39% complete
43% complete
Completing Database Creation
47% complete
49% complete
50% complete
Creating Pluggable Databases
54% complete
71% complete
Executing Post Configuration Actions
93% complete
Running Custom Scripts
100% complete
Database creation complete. For details check the logfiles at:
 /opt/oracle/cfgtoollogs/dbca/FREE.
Database Information:
Global Database Name:FREE
System Identifier(SID):FREE
Look at the log file "/opt/oracle/cfgtoollogs/dbca/FREE/FREE.log" for further details.

Connect to Oracle Database using one of the connect strings:
     Pluggable database: 9e2ad8cdd058/FREEPDB1
     Multitenant container database: 9e2ad8cdd058

SQL*Plus: Release 23.0.0.0.0 - Production on Tue May 14 08:44:22 2024
Version 23.4.0.24.05

Copyright (c) 1982, 2024, Oracle.  All rights reserved.


Connected to:
Oracle Database 23ai Free Release 23.0.0.0.0 - Develop, Learn, and Run for Free
Version 23.4.0.24.05

SQL>
System altered.

SQL>
Pluggable database altered.

SQL>
PL/SQL procedure successfully completed.

SQL> SQL>
Session altered.

SQL>
User created.

SQL>
Grant succeeded.

SQL>
Grant succeeded.

SQL>
Grant succeeded.

SQL>
User altered.

SQL> SQL> Disconnected from Oracle Database 23ai Free Release 23.0.0.0.0 - Develop, Learn, and Run for Free
Version 23.4.0.24.05
The Oracle base remains unchanged with value /opt/oracle
The Oracle base remains unchanged with value /opt/oracle
#####################################
########### E R R O R ###############
DATABASE SETUP WAS NOT SUCCESSFUL!
Please check output for further info!
########### E R R O R ###############
#####################################
The following output is now a tail of the alert.log:
ALTER SYSTEM SET local_listener='' SCOPE=BOTH;
ALTER PLUGGABLE DATABASE FREEPDB1 SAVE STATE
Completed: ALTER PLUGGABLE DATABASE FREEPDB1 SAVE STATE
2024-05-14T08:44:22.977086+00:00
Errors in file /opt/oracle/diag/rdbms/free/FREE/trace/FREE_ora_2733.trc:
ORA-07427: spstp: cannot change directory to dbs.
ORA-27300: OS system dependent operation:chdir() failed with status: 2
ORA-27301: OS failure message: No such file or directory
ORA-27302: failure occurred at: spstp:9
ORA-27303: additional information: error returned by chdir() is 2
2024-05-14T08:44:35.715092+00:00
Errors in file /opt/oracle/diag/rdbms/free/FREE/trace/FREE_j000_2737.trc:
ORA-07427: spstp: cannot change directory to dbs.
ORA-27300: OS system dependent operation:chdir() failed with status: 2
ORA-27301: OS failure message: No such file or directory
ORA-27302: failure occurred at: spstp:9
ORA-27303: additional information: error returned by chdir() is 2
2024-05-14T08:44:36.706783+00:00
kkjcre1p: unable to spawn jobq slave process
2024-05-14T08:44:36.707447+00:00
Errors in file /opt/oracle/diag/rdbms/free/FREE/trace/FREE_cjq0_2464.trc:
2024-05-14T08:44:36.997895+00:00
Errors in file /opt/oracle/diag/rdbms/free/FREE/trace/FREE_j000_2740.trc:
ORA-07427: spstp: cannot change directory to dbs.
ORA-27300: OS system dependent operation:chdir() failed with status: 2
ORA-27301: OS failure message: No such file or directory
ORA-27302: failure occurred at: spstp:9
ORA-27303: additional information: error returned by chdir() is 2
2024-05-14T08:44:37.989368+00:00
kkjcre1p: unable to spawn jobq slave process
2024-05-14T08:44:37.989589+00:00
Errors in file /opt/oracle/diag/rdbms/free/FREE/trace/FREE_cjq0_2464.trc:
2024-05-14T08:44:38.205260+00:00
Errors in file /opt/oracle/diag/rdbms/free/FREE/trace/FREE_j000_2742.trc:
ORA-07427: spstp: cannot change directory to dbs.
ORA-27300: OS system dependent operation:chdir() failed with status: 2
ORA-27301: OS failure message: No such file or directory
ORA-27302: failure occurred at: spstp:9
ORA-27303: additional information: error returned by chdir() is 2
2024-05-14T08:44:39.195527+00:00
kkjcre1p: unable to spawn jobq slave process
2024-05-14T08:44:39.195824+00:00
Errors in file /opt/oracle/diag/rdbms/free/FREE/trace/FREE_cjq0_2464.trc:
2024-05-14T08:44:39.411332+00:00
Errors in file /opt/oracle/diag/rdbms/free/FREE/trace/FREE_j000_2744.trc:
ORA-07427: spstp: cannot change directory to dbs.
ORA-27300: OS system dependent operation:chdir() failed with status: 2
ORA-27301: OS failure message: No such file or directory
ORA-27302: failure occurred at: spstp:9
ORA-27303: additional information: error returned by chdir() is 2
2024-05-14T08:44:40.401019+00:00
kkjcre1p: unable to spawn jobq slave process
2024-05-14T08:44:40.401222+00:00
Errors in file /opt/oracle/diag/rdbms/free/FREE/trace/FREE_cjq0_2464.trc:
2024-05-14T08:44:40.614321+00:00
Errors in file /opt/oracle/diag/rdbms/free/FREE/trace/FREE_j000_2746.trc:
ORA-07427: spstp: cannot change directory to dbs.
ORA-27300: OS system dependent operation:chdir() failed with status: 2
ORA-27301: OS failure message: No such file or directory
ORA-27302: failure occurred at: spstp:9
ORA-27303: additional information: error returned by chdir() is 2
```

**Comment by yahonda** [comment]
- Get the container id

```
$ docker ps
CONTAINER ID   IMAGE                         COMMAND                  CREATED         STATUS                            PORTS      NAMES
9e2ad8cdd058   oracle/database:23.4.0-free   "/bin/bash -c $ORACL…"   4 minutes ago   Up 4 minutes (health: starting)   1521/tcp   oracle-container
```
- Login to the container and get the trace file
```
$ docker exec -i -t 9e2ad8cdd058 /bin/bash
bash-4.4$ more /opt/oracle/diag/rdbms/free/FREE/trace/FREE_ora_2733.trc
Trace file /opt/oracle/diag/rdbms/free/FREE/trace/FREE_ora_2733.trc
Oracle Database 23ai Free Release 23.0.0.0.0 - Develop, Learn, and Run for Free
Version 23.4.0.24.05
Build label:    RDBMS_23.0.0.0.0_LINUX.X64_240424
ORACLE_HOME:    /opt/oracle/product/23ai/dbhomeFree
System name:	Linux
Node name:	9e2ad8cdd058
Release:	6.8.0-31-generic
Version:	#31-Ubuntu SMP PREEMPT_DYNAMIC Sat Apr 20 00:40:06 UTC 2024
Machine:	x86_64
Storage:	?
CLID:	U
Instance name: FREE
Instance number: 0 <none>
Redo thread mounted by this instance: 0 <none>
Oracle process number: 0
Unix process pid: 2733, NID: 4026532641, image:


*** 2024-05-14T08:44:22.976738+00:00
ORA-07427: spstp: cannot change directory to dbs.
ORA-27300: OS system dependent operation:chdir() failed with status: 2
ORA-27301: OS failure message: No such file or directory
ORA-27302: failure occurred at: spstp:9
ORA-27303: additional information: error returned by chdir() is 2
bash-4.4$ exit
```

**Comment by kaidonato** [comment]
Just to let you know. I'm still running into this issue. Let me know if I can assist.

**Comment by kaidonato** [comment]
@gvenzl 
This is the .trc

```
Trace file /opt/oracle/diag/rdbms/free/FREE/trace/FREE_ora_4214.trc
Oracle Database 23ai Free Release 23.0.0.0.0 - Develop, Learn, and Run for Free
Version 23.4.0.24.05
Build label:    RDBMS_23.0.0.0.0_LINUX.X64_240424
ORACLE_HOME:    /opt/oracle/product/23ai/dbhomeFree
System name:	Linux
Node name:	febd2b603ee1
Release:	4.18.0-519.el8.x86_64
Version:	#1 SMP Tue Oct 24 16:44:13 UTC 2023
Machine:	x86_64
VM name:	KVM CPUID feature flags: 0x01003afb
Storage:	?
CLID:	U
Instance name: FREE
Instance number: 0 <none>
Redo thread mounted by this instance: 0 <none>
Oracle process number: 0
Unix process pid: 4214, NID: 4026532646, image:


*** 2024-06-18T09:17:57.165801+00:00
ORA-07427: spstp: cannot change directory to dbs.
ORA-27300: OS system dependent operation:chdir() failed with status: 2
ORA-27301: OS failure message: No such file or directory
ORA-27302: failure occurred at: spstp:9
ORA-27303: additional information: error returned by chdir() is 2
sh-4.4$

```

**Comment by yunus-qureshi** [comment]
@kaidonato have you rebuilt after the fix was pushed?

**Comment by Sam-Hwang** [comment]
Got the same error in 23.5.0.24.07

Pulled the image container-registry.oracle.com/database/free:latest on 11/06/2024

Docker log
. . .
29% complete
Creating and starting Oracle instance
30% complete
33% complete
36% complete
39% complete
43% complete
Completing Database Creation
47% complete
49% complete
50% complete
Creating Pluggable Databases
54% complete
71% complete
Executing Post Configuration Actions
93% complete
Running Custom Scripts
100% complete
Database creation complete. For details check the logfiles at:
 /opt/oracle/cfgtoollogs/dbca/FREE.
Database Information:
Global Database Name:FREE
System Identifier(SID):FREE
Look at the log file "/opt/oracle/cfgtoollogs/dbca/FREE/FREE.log" for further details.

Connect to Oracle Database using one of the connect strings:
     Pluggable database: 21cda1142ee2/FREEPDB1
     Multitenant container database: 21cda1142ee2

SQL*Plus: Release 23.0.0.0.0 - for Oracle Cloud and Engineered Systems on Thu Nov 7 13:26:38 2024
Version 23.5.0.24.07

Copyright (c) 1982, 2024, Oracle.  All rights reserved.


Connected to:
Oracle Database 23ai Free Release 23.0.0.0.0 - Develop, Learn, and Run for Free
Version 23.5.0.24.07

SQL> 
System altered.

SQL> 
Pluggable database altered.

SQL> 
PL/SQL procedure successfully completed.

SQL> SQL> 
Session altered.

SQL> 
User created.

SQL> 
Grant succeeded.

SQL> 
Grant succeeded.

SQL> 
Grant succeeded.

SQL> 
User altered.

SQL> SQL> Disconnected from Oracle Database 23ai Free Release 23.0.0.0.0 - Develop, Learn, and Run for Free
Version 23.5.0.24.07

SQL*Plus: Release 23.0.0.0.0 - for Oracle Cloud and Engineered Systems on Thu Nov 7 13:26:39 2024
Version 23.5.0.24.07

Copyright (c) 1982, 2024, Oracle.  All rights reserved.


Connected to:
Oracle Database 23ai Free Release 23.0.0.0.0 - Develop, Learn, and Run for Free
Version 23.5.0.24.07

SQL> Database closed.
Database dismounted.
ORACLE instance shut down.
SQL> ORACLE instance started.

Total System Global Area 1603726632 bytes
Fixed Size		    5360936 bytes
Variable Size		  402653184 bytes
Database Buffers	 1191182336 bytes
Redo Buffers		    4530176 bytes
Database mounted.
SQL> 
Database altered.

SQL> 
Database altered.

SQL> 
Database altered.

SQL> SQL> Disconnected from Oracle Database 23ai Free Release 23.0.0.0.0 - Develop, Learn, and Run for Free
Version 23.5.0.24.07
The Oracle base remains unchanged with value /opt/oracle

Executing user defined scripts
/opt/oracle/runUserScripts.sh: running /opt/oracle/scripts/extensions/setup/registerService.sh

DONE: Executing user defined scripts

The Oracle base remains unchanged with value /opt/oracle
#####################################
########### E R R O R ###############
DATABASE SETUP WAS NOT SUCCESSFUL!
Please check output for further info!
########### E R R O R ###############
#####################################
The following output is now a tail of the alert.log:
FREEPDB1(3):Opening pdb with Resource Manager plan: DEFAULT_PLAN
Completed: Pluggable database FREEPDB1 opened read write 
Completed: ALTER DATABASE OPEN
2024-11-07T13:26:50.635657+00:00
Errors in file /opt/oracle/diag/rdbms/free/FREE/trace/FREE_ora_3311.trc:
ORA-07427: spstp: cannot change directory to dbs.
ORA-27300: OS system dependent operation:chdir() failed with status: 2
ORA-27301: OS failure message: No such file or directory
ORA-27302: failure occurred at: spstp:9
ORA-27303: additional information: error returned by chdir() is 2
2024-11-07T13:26:50.698139+00:00
